src/openvic-simulation/dataloader/Dataloader.cpp

```diff
@@ -824,6 +828,46 @@ bool Dataloader::_load_map_dir(DefinitionManager& definition_manager) const {
 	return ret;
 }
 
+bool Dataloader::_load_song_chances(DefinitionManager& definition_manager, fs::path song_chance_file) {
+	bool ret = true;
+	SongChanceManager& song_chance_manager = definition_manager.get_song_chance_manager();
+
+	if(song_chance_file.empty()){
+		Logger::info("No Songs.txt file to load");
+		song_chance_manager.lock_songChances();
+		return true;
+	}
+
+	ret &= song_chance_manager.load_songs_file(parse_defines_cached(song_chance_file).get_file_node());
```

Comment:
```suggestion
bool Dataloader::_load_song_chances(DefinitionManager& definition_manager) {
	static constexpr std::string_view song_chance_file = "music/songs.txt";

	bool ret = true;
	SongChanceManager& song_chance_manager = definition_manager.get_song_chance_manager();

	const fs::path song_chance_path = lookup_file(song_chance_file, false);

	if (song_chance_path.empty()) {
		Logger::info("No ", song_chance_file, " file to load");
		song_chance_manager.lock_song_chances();
		return true;
	}

	ret &= song_chance_manager.load_songs_file(parse_defines_cached(song_chance_path).get_file_node());
```
Since we'll always be loading the same file we can just have it as a constant at the top of this function instead of an argument.
Since we want to check if the file exists before attempting to load it, we can create a local `fs::path` variable to store it in. Calling `lookup_file` with `false` as the second argument prevents it from reporting an error when the file isn't found, needed for this case where the file might not exist without causing any problems.
We still use `parse_defines_cached` as we'll need to hold onto the parsers so we can parse the chance scripts after everything else is loaded.